qiskit/qasm3/__init__.py

```diff
@@ -370,3 +369,76 @@ def load_experimental(pathlike_or_filelike, /, *, custom_gates=None, include_pat
         category=ExperimentalWarning,
     )
     return _qasm3.load(pathlike_or_filelike, custom_gates=custom_gates, include_path=include_path)
+
+
+@functools.wraps(_qasm3.dumps)
+def dumps_experimental(
+    circuit,
+    *,
+    includes=None,
+    basis_gates=None,
+    disable_constants=True,
+    alias_classical_registers=False,
+    allow_aliasing=False,
+    indent="  ",
+):
+    """<overridden by functools.wraps>"""
+    warnings.warn(
+        "This is an experimental version of serialization of a :class:`~qiskit.circuit.QuantumCircuit`"
+        " object in an OpenQASM 3 string."
+        " Beware that its interface might change, and it might be missing features.",
+        category=ExperimentalWarning,
+    )
+    if includes is None:
+        includes = ["stdgates.inc"]
+    if basis_gates is None:
+        basis_gates = ["U"]
+    return _qasm3.dumps(
+        circuit,
+        {
+            "includes": includes,
+            "basis_gates": basis_gates,
+            "disable_constants": disable_constants,
+            "alias_classical_registers": alias_classical_registers,
+            "allow_aliasing": allow_aliasing,
+            "indent": indent,
+        },
+    )
```

Comment:
This is the strange calling convention I mentioned in the previous review comment - it'd be more natural if these were just regular keyword arguments, even if it is just an internal function.